CMakeLists.txt

```diff
@@ -235,10 +235,21 @@ IF (WIN32 AND EMBED_SSH_PATH)
 	ADD_DEFINITIONS(-DGIT_SSH)
 ENDIF()
 
-IF (WIN32 AND WINHTTP)
-	ADD_DEFINITIONS(-DGIT_WINHTTP)
+
+FIND_PACKAGE(HTTP_Parser)
+IF (HTTP_PARSER_FOUND AND HTTP_PARSER_VERSION_MAJOR EQUAL 2)
```

Comment:
I'd say this check `HTTP_PARSER_VERSION_MAJOR EQUAL 2` doesn't quite match up with the message for when it fails. `http-parser was not found or is too old; using bundled 3rd-party sources.` since the failure could (in the future) be due to a version 3 being released?

But this was the case in the old version too, so maybe it doesn't matter.